packages/react-native-worklets/src/shareables.ts

```diff
@@ -163,6 +163,11 @@ function makeShareableCloneRecursiveNative<T>(
     return cloneRemoteFunction(value, shouldPersistRemote);
   }
   if (isHostObject(value)) {
+    // RN has introduced a new representation of the turbo module as a JS object whose prototype is the host object
+    // More details: https://github.com/facebook/react-native/blob/main/packages/react-native/ReactCommon/react/nativemodule/core/ReactCommon/TurboModuleBinding.cpp#L182
+    if (isHostObject(Object.getPrototypeOf(value))) {
```

Comment:
How is this going to work? I thought the object itself it a plain JS object and only its prototype is a HostObject, whereas here `if (isHostObject(Object.getPrototypeOf(value)))` is inside `if (isHostObject(value))`. How does it work then if both conditions need to be met?

Reply:
if the prototype of an object is a `HostObject ` then the object itself is also a `HostObject`, But to improve code readability, I'll move this check above (outside `if (isHostObject(value))`)